cpp/src/arrow/filesystem/filesystem.h

```diff
@@ -128,12 +128,14 @@ struct ARROW_EXPORT FileSelector {
   /// The behavior if `base_dir` isn't found in the filesystem.  If false,
   /// an error is returned.  If true, an empty selection is returned.
   bool allow_not_found;
+  /// The behavior if errors occur such as access denied during directory traversal.
+  bool allow_errors;
```

Comment:
Perhaps this should be more explicitly called `bool ignore_permission_errors` or something similar?